compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEPad.cpp

```diff
@@ -19,108 +19,106 @@ namespace mlir::iree_compiler::AMDAIE {
 
 namespace {
 
-static FailureOr<linalg::LinalgPaddingOptions>
-getFirstLevelLinalgPaddingOptions(IRRewriter &rewriter,
-                                  linalg::LinalgOp &linalgOp) {
-  linalg::LinalgPaddingOptions options;
+static SmallVector<Attribute> getPaddingValues(IRRewriter &rewriter,
+                                               linalg::LinalgOp &linalgOp) {
   SmallVector<Attribute> paddingValues;
   for (auto operand : linalgOp->getOperands()) {
-    auto type = dyn_cast<RankedTensorType>(operand.getType());
-    if (!type) {
-      linalgOp->emitOpError("expected ranked tensor type");
-      return failure();
+    Type elementType;
+    if (auto type = dyn_cast<RankedTensorType>(operand.getType())) {
+      elementType = type.getElementType();
+    } else if (isa<arith::ConstantOp>(operand.getDefiningOp())) {
+      elementType = operand.getType();
     }
-    Type elementType = type.getElementType();
+
     if (isa<IntegerType>(elementType)) {
       paddingValues.push_back(rewriter.getIntegerAttr(elementType, 0));
     } else {
       paddingValues.push_back(rewriter.getFloatAttr(elementType, 0));
     }
   }
-  options.paddingValues = paddingValues;
-  SmallVector<bool> packPaddings(linalgOp->getNumOperands(), true);
-  options.packPaddings = packPaddings;
+  return paddingValues;
+}
 
+static SmallVector<int64_t> getPaddingDimensions(linalg::LinalgOp &linalgOp) {
   SmallVector<int64_t> paddingDimensions;
-  for (unsigned i = 0, n = linalgOp.getNumLoops(); i < n; i++)
+  for (unsigned i = 0; i < linalgOp.getNumLoops(); i++) {
```

Comment:
I'd prefer the first syntax.

Any particular reason for the new change being suggested?

Reply:
It looks cleaner to me.